enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,284 @@
+    MaxDeviceCount int `json:"maxDeviceCount"`
+    // VolumeMode determines whether the PV created is Block or Filesystem
+    // + optional
+    VolumeMode PersistentVolumeMode `json:"volumeMode,omitempty"`
+    // FSType type to be used when volumeMode is Filesystem
+    // +optional
+    FSType string `json:"fsType,omitempty"`
+    // DeviceInclusionSpec are the filtration rules for including a device in the device discovery
+    // +optional
+    DeviceInclusionSpec *DeviceInclusionSpec `json:"deviceInclusionSpec"`
+}
+
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational, nonRotational or
+// RotationalAndNonRotational
+type DeviceMechanicalProperty string
+
+const (
+    // The mechanical properties of the devices
+    // Rotational refers to magnetic disks
+    Rotational DeviceMechanicalProperty = "rotational"
+    // NonRotational refers to ssds
+    NonRotational DeviceMechanicalProperty = "nonRotational"
+    // RotationalAndNonRotational refers to both magnetic and ssds
+    RotationalAndNonRotational DeviceMechanicalProperty = ""
+)
+
+type DeviceDiscoveryPolicyType string
+
+const (
+    // The DeviceDiscoveryPolicies that will be supported by the LSO.
+    // These Discovery policies will based on lsblk's type output
+    // Disk represents a device-type of disk
+    Disk DeviceDiscoveryPolicyType = "disk"
+    // Part represents a device-type of partion
+    Part DeviceDiscoveryPolicyType = "part"
+)
+
+type DeviceInclusionSpec struct {
+    // DeviceTypes that should be used for automatic detection. This would be one of the types supported
+    // by the local-storage operator. Currently the supported types are: disk,part
+    // If the list is empty the default value will be `[disk]`.
+    DeviceTypes []DeviceDiscoveryPolicyType `json:"deviceType"`
+
+    // DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+    // by default, RotationalAndNonRotational which matches all disks.
+    // +optional
+    DeviceMechanicalProperty DeviceMechanicalProperty `json:"deviceMechanicalProperty"`
+
+    // MinSize is the minimum size of the device which needs to be included
+    // +optional
+    MinSize resource.Quantity `json:"minSize"`
+
+    // MaxSize is the maximum size of the device which needs to be included
+    // +optional
+    MaxSize resource.Quantity `json:"maxSize"`
+
+    // Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+    // to contain at least one of these strings.
+    // +optional
+    Models []string `json:"models"`
+    
+    // Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+    // to contain at least one of these strings.
+    // +optional
+    Vendors []string `json:"vendors"`
+}
+```
+
+some examples of autoDetectVolume CR instance:
+
+```yaml
+apiVersion: local.storage.openshift.io/v1
```

Comment:
Lets designate this new type as `v1alpha` or something similar.

Reply:
done